consensus/core/src/hashing/sighash.rs

```diff
@@ -0,0 +1,258 @@
+use hashes::{Hash, Hasher, HasherBase, TransactionSigningHash};
+
+use crate::{
+    subnets::SUBNETWORK_ID_NATIVE,
+    tx::{PopulatedTransaction, ScriptPublicKey, TransactionOutpoint, TransactionOutput},
+};
+
+use super::{sighash_type::SigHashType, HasherExtensions};
+
+#[derive(Default)]
+pub struct SigHashReusedValues {
+    previous_outputs_hash: Option<Hash>,
+    sequence_hash: Option<Hash>,
+    sig_op_counts_hash: Option<Hash>,
+    outputs_hash: Option<Hash>,
+}
+
+impl SigHashReusedValues {
+    pub fn new() -> Self {
+        Self { previous_outputs_hash: None, sequence_hash: None, sig_op_counts_hash: None, outputs_hash: None }
+    }
+}
+
+fn previous_output_hash(tx: &PopulatedTransaction, hash_type: SigHashType, reused_values: &mut SigHashReusedValues) -> Hash {
+    if hash_type.is_sighash_anyone_can_pay() {
+        return 0.into();
```

Comment:
I think we should replace all `0.into()` in this file with `ZERO_HASH` (defined in `hashes::`)